libchisel/src/checkstartfunc.rs

```diff
@@ -15,7 +15,7 @@ impl CheckStartFunc {
 }
 
 impl ModuleValidator for CheckStartFunc {
-    fn validate(&self, module: &Module) -> Result<bool, String> {
+    fn validate(&self, module: &Module) -> Result<bool, ModuleError> {
```

Comment:
Alternative idea: Replace the current `Result<bool, ModuleError>` with a simpler `ModuleResult` which can have the variants `Valid, Invalid, Mutated, Unsupported, Other(String)`

Reply:
@axic what do you think about this?

Reply:
One downside is that I think `unwrap` cannot be used in that case.

Reply:
Oh yeah, true. I just realized that the methods are implemented directly rather than via a trait. this is good for merge then?